From 517d0f64ac1fb702cd85eefbf73bdd849dd07012 Mon Sep 17 00:00:00 2001
From: jurulz <60669313+jurulz@users.noreply.github.com>
Date: Sun, 18 Oct 2026 14:20:22 -0400
Subject: [PATCH 1/6] Common: Fix GetLocalTimeSinceJan1970 on Windows 10 before
 1903

std::chrono::current_zone() needs ICU, which only exists on Windows 10 1903 and later, and throws on older builds (for example 1809 LTSC used by many cabinet PCs). Use SystemTimeToTzSpecificLocalTime on Windows instead.
---
 Source/Core/Common/Timer.cpp | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/Source/Core/Common/Timer.cpp b/Source/Core/Common/Timer.cpp
index fcbbd7b..32789b3 100644
--- a/Source/Core/Common/Timer.cpp
+++ b/Source/Core/Common/Timer.cpp
@@ -68,11 +68,21 @@ u64 Timer::ElapsedMs() const
 
 u64 Timer::GetLocalTimeSinceJan1970()
 {
-#ifdef _MSC_VER
-  std::chrono::zoned_seconds seconds(
-      std::chrono::current_zone(),
-      std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()));
-  return seconds.get_local_time().time_since_epoch().count();
+#ifdef _WIN32
+  // std::chrono::current_zone() needs ICU, which only exists on Windows 10 1903+.
+  // Use the Win32 API so this also works on older Windows 10 builds.
+  SYSTEMTIME utc_time{};
+  SYSTEMTIME local_time{};
+  GetSystemTime(&utc_time);
+  if (!SystemTimeToTzSpecificLocalTime(nullptr, &utc_time, &local_time))
+    local_time = utc_time;
+  FILETIME file_time{};
+  SystemTimeToFileTime(&local_time, &file_time);
+  ULARGE_INTEGER ticks;
+  ticks.LowPart = file_time.dwLowDateTime;
+  ticks.HighPart = file_time.dwHighDateTime;
+  // FILETIME counts 100ns intervals since 1601-01-01.
+  return ticks.QuadPart / 10000000ULL - 11644473600ULL;
 #else
   time_t sysTime, tzDiff, tzDST;
   time(&sysTime);
-- 
2.54.0.windows.1

